session/session.go

```diff
@@ -965,14 +965,23 @@ func (s *session) ParseSQL(ctx context.Context, sql, charset, collation string)
 }
 
 func (s *session) SetProcessInfo(sql string, t time.Time, command byte, maxExecutionTime uint64) {
+	var db interface{}
+	if len(s.sessionVars.CurrentDB) > 0 {
+		db = s.sessionVars.CurrentDB
+	}
+
+	var info interface{}
```

Comment:
How about not changing the type of `Info` from `string` to `interface{}`. We can do it this way:
```go
info := "NULL"
if len(sql) > 0 {
	info = sql
}
```

Reply:
Never mind. Could you add a test like:
```sql
select * from information_schema.processlist where db is null;
```